cylc/flow/parsec/validate.py

```diff
@@ -204,10 +204,28 @@ def validate(self, cfg_root, spec_root):
                 else:
                     speckey = key
                 specval = spec[speckey]
-                if isinstance(value, dict) and not specval.is_leaf():
+
+                cfg_is_section = isinstance(value, dict)
+                spec_is_section = not specval.is_leaf()
+                if cfg_is_section and not spec_is_section:
+                    # config is a [section] but it should be a setting=
+                    raise IllegalItemError(
+                        keys,
+                        key,
+                        msg=f'"{key}" should be a setting not a [section]',
+                    )
+                if not cfg_is_section and spec_is_section:
```

Comment:
Like when an Oxford comma is missing, my brain doesn't like the ambiguity
```suggestion
                if (not cfg_is_section) and spec_is_section:
```
or
```suggestion
                if spec_is_section and not cfg_is_section:
```

Reply:
I'll parenthesise it (because I like brackets), but I wouldn't *expect* anyone to, at least for simple expressions like this. It's Python's bidmas, and & or bind more tightly than not.

git grep 'if not.*and'

Reply:
I know but sometimes it's nice to use parentheses in maths too for ease of reading e.g. `6 / 2 * 3` vs `(6 / 2) * 3`